docs/sources/upgrading/_index.md

```diff
@@ -31,6 +31,45 @@ The output is incredibly verbose as it shows the entire internal config struct u
 
 ## Main / Unreleased
 
+### Loki
+
+#### `stream_lag_labels` configuration has moved, single client config has been deprecated
+
+Previously the config for each promtail client looked like: https://grafana.com/docs/loki/v2.4/clients/promtail/configuration/#clients
+example:
+```
+clients:
+  - url: http://localhost:3100/loki/api/v1/push
+    stream_lag_labels: "filename,something_else_1"
+    external_labels:
+      client: a
+  - url: http://localhost:3100/loki/api/v1/push
+    stream_lag_labels: "filename,something_else_2"
+    external_labels:
+      client: b
+```
+However due to the way the Promtail client metrics were registered, and more specifically the way the Promtail stream lag metric labels were set, there was a bug
+that should have been caught by Prometheus' metric instrumentation library (duplicate metric registration/attempt to register the same metric with different label sets)
+but was not, and therefore would result in a failure to scrape metrics at runtime. This was due to allowing `stream_lag_labels` to be set for each client,
+which could result in that metric for each client having the same label values. To remedy this, the `stream_lag_labels` config field has been moved to a new
+top level section `options`.
+example:
```

Comment:
This is a bit hard to read, mostly because of the length of the sentences.  I'm suggesting changes that make the read more choppy, but hopefully more clear.

```suggestion
A failure to scrape metrics at runtime was caused by the combination of the way the Promtail client metrics were registered and the Prometheus metric instrumentation library not catching a bug.
The duplicate metric registration/attempt to register the same metric with different label sets allowed `stream_lag_labels` to be set for each client,
which could result in that metric for each client having the same label values.
The remedy moves the `stream_lag_labels` parameter configuration to a new
top-level `options` configuration block.

If you currently set a value for configuration per client, you will need to move and only set it once, as in this example:

Reply:
Thanks for the review @KMiller-Grafana, some of the changes in your suggestion changed the meaning of the cause of the bug (at least how I interpreted it) so I made a few edits to this section in my commit.